apiserver/facades/client/modelmanager/modelinfo_test.go

```diff
@@ -439,6 +463,15 @@ func (s *modelInfoSuite) TestModelInfoWriteAccess(c *gc.C) {
 	s.mockMachineService.EXPECT().InstanceIDAndName(gomock.Any(), "deadbeef2").Return("inst-deadbeef2", "", nil)
 	s.mockMachineService.EXPECT().HardwareCharacteristics(gomock.Any(), "deadbeef1").Return(&instance.HardwareCharacteristics{}, nil)
 
+	modelInfoService := mocks.NewMockModelInfoService(ctrl)
+	modelInfoService.EXPECT().GetModelInfo(gomock.Any()).Return(coremodel.ReadOnlyModel{
+		AgentVersion:   jujuversion.Current,
+		ControllerUUID: s.controllerUUID,
+		Cloud:          "dummy",
+		CloudType:      "dummy",
+	}, nil)
+	s.mockModelDomainServices.EXPECT().ModelInfo().Return(modelInfoService)
```

Comment:
Could this line specifically go in test setup? Since it seems to be the same every time
```
modelDomainServices.EXPECT().ModelInfo().Return(modelInfoService)
```

Reply:
No, not all callers of the getAPIWithUser setup method need the modelInfoService mocked.

Reply:
`.AnyTimes()` would solve that. However it's not worth arguing about, I won't push it any further.